### DISCUSSION TITLE
Add a differen "State" for solved issues vs closed issues

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#4664] [Projects and Issues] [answered]
## The Problem
Issues generally have two states they can be in: `Open` and `Closed`

This may not be problematic at first, but when you look at closed issues in a list is there not really any indicator of if the issue itself was addressed and solved or just closed. At most could there be labels stating this, but that is obviously not always the case for repositories.

Additionally are the colours used for the Open and Closed states a bit misused imo. Closed is shown in Red which I always associate with a bad thing, as in "It's bad that this issue was closed" while I see green as a good thing, which may be the case for an issue when it reports a critical bug or suggests a useful feature.

## The idea
Issues should adopt a system similar to Pull requests. A Pull request can have 4 states: `Draft`, `Open`, `Closed` and `Merged`. What is interesting to note is, that both `Closed` and `Merged` result in the Pull request itself being closed and the only difference being if the changes were merged into the branch or not.

Maybe Issues can have a similar thing where they have a 3rd state called `Solved`.
That way could a maintainer, or the Issue-creator indicate that the issue has been solved, while closing it would indicate that it was just closed without any real solution or similar.
An option to change the Solved state of a closed issue should then also be provided.

In addition, should it be reconsidered what colours are used for the states of the Issue as Red is not really a matching colour for an issue that has been solved through a Pull request or similar?

## Alternatives
One alternative I can think of is to use Labels to indicate whether an issue was solved or not, but this feels like a semi-optimal solution to have here.

## Closing words
I hope this suggestion makes sense. I posted it in the Issues Feedback category since it fits there the best. If not, can I move it to general feedback.

**Comment by Zerotask** [comment]
That's very similar to my request: https://github.com/github/feedback/discussions/4514

**Reply by leighjboyd** [reply]
I could not agree more.  In fact, I want at least 10 different states to represent the entire design (not just bugs).  Im new to Git, so please tell me if I'm not thinking about this right (maybe I should just use labels?) But I want something more like the following:

new  (just a stub, not fully explained issue yet)-> open (fully defined)  -> 
assigned (now we know whos best to handle it)->
-> in progress ->  blocked (optional) -> ready for test/review ->  in review -> closed 

also like you say, more granularity in the "why is it closed" category
closed - rejected (the original report was in error)
closed - with explanation (in other words nothing changed in the code, perhaps documentation was fixed or requirement was relaxed, etc)
closed - duplicate 
closed - parked (no time for this according to PM!)


The main problem is that you have developers which call things closed when they have not yet even been committed, tested, documented, etc etc.  so this granularity is useful to see progress at a glance.  Instead of having hundreds of open items lying about with no easy way to filter the important from the lower priority ones....

**Reply by Andre601** [reply]
For this kind of granularity would projects be better suited as you would assign the issue to a project and move that card from open to "to do" -> "awaiting review" -> "done" or whatever setup you prefer.

**Reply by leighjboyd** [reply]
Ha! I just discovered Projects (beta) now, it seems like its exactly what I was looking for, thanks!

**Reply by EmilyRosina** [reply]
Still searching for docs for this, but it seems that it has been implemented :smile_cat: 

![image](https://user-images.githubusercontent.com/21045073/170537239-8d983b73-5b3e-4dd6-8470-d282686e3986.png)

**Comment by azenMatt** [comment, marked as answer]
Hi 👋 the new issue closed reasons - https://github.blog/changelog/2022-05-19-the-new-github-issues-may-19th-update/ - give you additional options for communicating why issues were closed

**Reply by waterplea** [reply]
Adding `reason:complete` does not work:
https://github.com/taiga-family/taiga-ui/issues?q=reason%3Acomplete

**Reply by EmilyRosina** [reply]
It's `reason:completed` not `reason:complete` :see_no_evil:  

I checked the docs for [filtering issues and PRs](https://docs.github.com/en/issues/tracking-your-work-with-issues/filtering-and-searching-issues-and-pull-requests#:~:text=Filter%20issues%20by%20the%20reason%20they%20were%20closed%3A%20is%3Aclosed%20reason%3Acompleted%20or%20is%3Aclosed%20reason%3A%22not%20planned%22) and [filtering projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/customizing-views-in-your-project/filtering-projects#filtering-by-close-reason), everything looks ok there.
I found `state_reason` via the [api docs](https://docs.github.com/en/rest/issues/issues?apiVersion=2022-11-28#:~:text=state_reason%20string%20or,reopened%2C%20null), which gave me an exhaustive list of valid values: 
- `reason:` `completed`
- `reason:` `not_planned` _(`not-planned` & `"not planned"` also work)_
- `reason:` `reopened`
- `reason:` `null`

These should all work for you :smile_cat: 
- [`reason:not_planned`](https://github.com/taiga-family/taiga-ui/issues?q=reason%3Anot_planned) 
- [`reason:not-planned`](https://github.com/taiga-family/taiga-ui/issues?q=reason%3Anot-planned)
- [`reason:"not planned"`](https://github.com/taiga-family/taiga-ui/issues?q=reason%3A"not+planned") 
- [`reason:completed`](https://github.com/taiga-family/taiga-ui/issues?q=reason%3Acompleted)
- [`reason:reopened`](https://github.com/taiga-family/taiga-ui/issues?q=reason%3Areopened)
- [`reason:null`](https://github.com/taiga-family/taiga-ui/issues?q=reason%3Anull)

**Reply by ljcool2006** [reply]
> It's `reason:completed` not `reason:complete` 🙈
> 
> I checked the docs for [filtering issues and PRs](https://docs.github.com/en/issues/tracking-your-work-with-issues/filtering-and-searching-issues-and-pull-requests#:~:text=Filter%20issues%20by%20the%20reason%20they%20were%20closed%3A%20is%3Aclosed%20reason%3Acompleted%20or%20is%3Aclosed%20reason%3A%22not%20planned%22) and [filtering projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/customizing-views-in-your-project/filtering-projects#filtering-by-close-reason), everything looks ok there. I found `state_reason` via the [api docs](https://docs.github.com/en/rest/issues/issues?apiVersion=2022-11-28#:~:text=state_reason%20string%20or,reopened%2C%20null), which gave me an exhaustive list of valid values:
> 
>     * `reason:` `completed`
> 
>     * `reason:` `not_planned` _(`not-planned` & `"not planned"` also work)_
> 
>     * `reason:` `reopened`
> 
>     * `reason:` `null`
> 
> 
> These should all work for you 😸
> 
>     * [`reason:not_planned`](https://github.com/taiga-family/taiga-ui/issues?q=reason%3Anot_planned)
> 
>     * [`reason:not-planned`](https://github.com/taiga-family/taiga-ui/issues?q=reason%3Anot-planned)
> 
>     * [`reason:"not planned"`](https://github.com/taiga-family/taiga-ui/issues?q=reason%3A%22not+planned%22)
> 
>     * [`reason:completed`](https://github.com/taiga-family/taiga-ui/issues?q=reason%3Acompleted)
> 
>     * [`reason:reopened`](https://github.com/taiga-family/taiga-ui/issues?q=reason%3Areopened)
> 
>     * [`reason:null`](https://github.com/taiga-family/taiga-ui/issues?q=reason%3Anull)

how to show completed issues but not "not planned" and duplicate issues?

**Comment by fabriziobitti** [comment]
The issue can have different states depending on how you organize your team. if you have QA team, they will go on Fixed issues and put it on closed only if verified. at the same time the developers can check the open issues and fix it putting them is state "fixed". this is they way it should work. knowing the reason for closing is not a valuable info, that should be in the comments

**Comment by Andre601** [comment]
Comming back to this discussion, I recently noticed a smaller thing with the issue states, which is, that you cannot mark an issue as simply "invalid".

The close states issues allow right now is completed, not planned and duplicate (With last one also allowing an issue number to be defined).

But more than once do you have situations where an issue simply isn't valid, be it because the template wasn't followed, because an unsupported/illegitimate version of a software or similar was used, or something else.
Being able to mark an issue as simply being invalid would be a great help here as "not planned" doesn't really feel like the best option here to use.

I may open a new discussion for this, if that is something worth considering.
